### PR TITLE
POC 2: Creating a flat component structure

### PR DESCRIPTION
**Pull request by brandenrodgers**
The main goal of this POC is to make project components as atomic/modular as possible. It removes the hierarchical structure of an app and its subcomponents in favor of a flatter overall project structure. Each subcomponent moves up one layer and becomes its own top-level component. We trade hierarchy for inter-component dependencies. This enables auth-reliant components to continue to work the same way that they work today.

### CODE REVIEW

**Comment by brandenrodgers** [comment]
I'm going to merge this so that it shows up in the main branch, but everyone should feel free to continue to comment on this PR!

File: POC 2 (flat components)/poc-2-project/hsproject.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "POC 2 project",
```
Comment (m-roll) [inline POC 2 (flat components)/poc-2-project/hsproject.json:2]: verbiage thought: what about this config is "flat"? Is there a better term we could use?

File: POC 2 (flat components)/poc-2-project/src/auth/hscomponent.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "type": "app",
```
Comment (m-roll) [inline POC 2 (flat components)/poc-2-project/src/auth/hscomponent.json:2]: I like this -- the name and UID are required, and the data is the "metadata" around this component.

File: POC 2 (flat components)/poc-2-project/src/card1/hscomponent.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "type": "crm-card",
+  "name": "POC 2 Card 1",
+  "uid": "poc_2_card_1",
+  "auth": "poc_2_app",
```
Comment (m-roll) [inline POC 2 (flat components)/poc-2-project/src/card1/hscomponent.json:5]: Wonder if we could/should leave it open to the developer to inline the `auth` field here.

**Review by m-roll — COMMENTED** [review]
some thoughts.

File: POC 2 (flat components)/poc-2-project/src/card1/hscomponent.json (hunk 3 above)
Comment (m-roll) [reply POC 2 (flat components)/poc-2-project/src/card1/hscomponent.json:5]: I do like this though, the reference with `uid` is neat.

**Pull request merged**